ex2_services/Services/source/housekeeping/housekeeping_service.c

```diff
@@ -573,18 +573,7 @@ Result write_hk_to_file(uint16_t filenumber, All_systems_housekeeping *all_hk_da
     red_lseek(fout, (filenumber - 1) * needed_size, RED_SEEK_SET);
 
     red_errno = 0;
-    /*The order of writes and subsequent reads must match*/
-    red_write(fout, &all_hk_data->hk_timeorder, sizeof(all_hk_data->hk_timeorder));
-    red_write(fout, &all_hk_data->adcs_hk, sizeof(all_hk_data->adcs_hk));
-    red_write(fout, &all_hk_data->Athena_hk, sizeof(all_hk_data->Athena_hk));
-    red_write(fout, &all_hk_data->EPS_hk, sizeof(all_hk_data->EPS_hk));
-    red_write(fout, &all_hk_data->UHF_hk, sizeof(all_hk_data->UHF_hk));
-    red_write(fout, &all_hk_data->S_band_hk, sizeof(all_hk_data->S_band_hk));
-    red_write(fout, &all_hk_data->hyperion_hk, sizeof(all_hk_data->hyperion_hk));
-    red_write(fout, &all_hk_data->charon_hk, sizeof(all_hk_data->charon_hk));
-    red_write(fout, &all_hk_data->DFGM_hk, sizeof(all_hk_data->DFGM_hk));
-    red_write(fout, &all_hk_data->NS_hk, sizeof(all_hk_data->NS_hk));
-    red_write(fout, &all_hk_data->IRIS_hk, sizeof(all_hk_data->IRIS_hk));
+    red_write(fout, all_hk_data, sizeof(All_systems_housekeeping));
```

Comment:
The seek used "needed_size", but the write uses sizeof(All_systems_housekeeping). Are they the same?

Reply:
I'm not sure what that's about. Maybe I should give it a proper review